### ISSUE TITLE
bug: Changing a Google User's primary email leads to uncaught exception

### ISSUE THREAD

**Issue by mat-release** [slashdevops/idp-scim-sync#57]
**Describe the bug**

A Google Workspace can have multiple Gmail domains, e.g. `@foo.com` and `@bar.com`. If a user's email domain is changed after previously synced to AWS SSO via SCIM, then a subsequent SSO sync attempts to create a new user  leads to an error.

```
WARN[0012] creating users                                quantity=3
WARN[0012] aws checkHTTPResponse: body: {"schema":["urn:ietf:params:scim:api:messages:2.0:Error"],"schemas":["urn:ietf:params:scim:api:messages:2.0:Error"],"detail":"Refused to create a new, duplicate resource.","status":"409","exceptionRequestId":"1f0de3fa-8bce-49ed-ae22-8604172fcb80","timeStamp":"2022-03-22 01:39:54.152"}  method=POST status="409 Conflict"
Error: cannot sync groups and their members: error syncing state: error reconciling users: error creating users in SCIM Provider: scim: error creating user: aws CreateUser: error decoding response body: EOF

Error: cannot sync groups and their members: error syncing state: error reconciling users: error creating users in SCIM Provider: scim: error creating user: aws CreateUser: error decoding response body: EOF
```

**Afterward**, if I re-run the SSO sync script, a **new** error appears. This time, the program attempts to duplicate the group (not the user):

```
INFO[0011] provider groups and state groups are different 
INFO[0011] reconciling groups                            idp=4 state=3
WARN[0011] creating groups                               quantity=1
WARN[0011] creating group                                group="AWS SSO - Shared Sandbox - Full Admin"
WARN[0011] aws checkHTTPResponse: body: {"schema":["urn:ietf:params:scim:api:messages:2.0:Error"],"schemas":["urn:ietf:params:scim:api:messages:2.0:Error"],"detail":"Refused to create a new, duplicate resource.","status":"409","exceptionRequestId":"80922f78-7149-4cf6-bf7c-4748047072dd","timeStamp":"2022-03-22 01:40:31.832"}  method=POST status="409 Conflict"
Error: cannot sync groups and their members: error syncing state: error reconciling groups: error creating groups in SCIM Provider: scim: error creating group: aws CreateGroup: error decoding response body: %!w(<nil>), body: 
```

This is actually strange, as nothing about the group has changed other than the email of a member. Maybe that's all that's needed?

**To Reproduce**

1. [Set up Google Workspace with two Gmail domains]
1. Create user for Mat, e.g. `mat@foo.com`
1. Create any group, e.g. `aws-sso-test` and add Mat to the group
1. Sync group to AWS SSO 
1. Change Mat's gmail / primary email domain from `mat@foo.com` to `mat@bar.com`
1. Attempt to sync again
1. **Error**
1. Delete group and user from AWS SSO
1. Run sync again, this time everything worked!

**Comment by christiangda** [comment]
Hi, @mat-release thank you for reporting that.  Which version of the software are you using? the latest one?

**Comment by christiangda** [comment]
@mat-release this was fixed in #45, release [v0.0.10 ](https://github.com/slashdevops/idp-scim-sync/releases/tag/v0.0.10).  Could you confirm that?
